### ISSUE TITLE
Not only for Markdown

### ISSUE THREAD

**Issue by EdJoPaTo** [dundalek/markmap#9]
Did you thought about generalizing this library in order to not work only with markdown?

The [atom package](https://github.com/dundalek/atom-markdown-mindmap) including this library is the only mindmapping tool i found for atom and has potential.

I thought about a line based mind map description language and did not found something ease to write and well versionizable with git. So i just implemented an own parser (#8) and that worked well. I could do that for LaTeX headers (sections and so on) too but I'm not that familiar with your atom package source code in order to include that.

**Comment by dundalek** [comment]
Yeah, it definitely crossed my mind. There is a prototype [parse.pandoc.js](https://github.com/dundalek/markmap/blob/master/parse.pandoc.js) that uses [pandoc](http://pandoc.org/) which supports tons of formats. I did not get to integrating it into the atom package because I haven't found a reliable way to include pandoc as a dependency.

**Comment by seifer08ms** [comment]
@dundalek ,I have a thought that we could integrate it into Rmarkdown in Rstudio which have already included pandoc as a dependency. So we can render a Rmarkdown file into a mindmap format such as html,pdf or doc.

**Comment by dundalek** [comment]
@seifer08ms If Rstudio supports javascript modules it should be possible to create such extension. Markmap is designed to be easily embedded, I use it in web apps as well as an extension for Atom editor.
